Sources/_MatchingEngine/Regex/Parse/CharacterPropertyClassification.swift

```diff
@@ -0,0 +1,395 @@
+extension Source {
+  static private func normalizeForCharacterPropertyMatching(
+    _ str: String
+  ) -> String {
+    // This follows the rules provided by UAX44-LM3, including the dropping of
+    // an "is" prefix, which isn't required by UTS#18 RL1.2, but is nice for
+    // consistency with other engines and the Unicode.Scalar.Properties names.
+    let str = str.filter { !$0.isWhitespace && $0 != "_" && $0 != "-" }
+                 .lowercased()
+    if str.hasPrefix("is") {
+      return String(str.dropFirst(2))
+    }
+    return str
+  }
+
+  static private func classifyGeneralCategory(
+    _ str: String
+  ) -> Unicode.ExtendedGeneralCategory? {
+    // This uses the aliases defined in
+    // https://www.unicode.org/Public/UCD/latest/ucd/PropertyValueAliases.txt.
+    switch normalizeForCharacterPropertyMatching(str) {
+    case "c", "other": return .other
+    case "cc", "control", "cntrl": return .control
+    case "cf", "format": return .format
+    case "cn", "unassigned": return .unassigned
+    case "co", "privateuse": return .privateUse
+    case "cs", "surrogate": return .surrogate
+    case "l", "letter": return .letter
+    case "lc", "casedletter": return .casedLetter
+    case "ll", "lowercaseletter": return .lowercaseLetter
+    case "lm", "modifierletter": return .modifierLetter
+    case "lo", "otherletter": return .otherLetter
+    case "lt", "titlecaseletter": return .titlecaseLetter
+    case "lu", "uppercaseletter": return .uppercaseLetter
+    case "m", "mark", "combiningmark": return .mark
+    case "mc", "spacingmark": return .spacingMark
+    case "me", "enclosingmark": return .enclosingMark
+    case "mn", "nonspacingmark": return .nonspacingMark
+    case "n", "number": return .number
+    case "nd", "decimalnumber", "digit": return .decimalNumber
+    case "nl", "letternumber": return .letterNumber
+    case "no", "othernumber": return .otherNumber
+    case "p", "punctuation", "punct": return .punctuation
+    case "pc", "connectorpunctuation": return .connectorPunctuation
+    case "pd", "dashpunctuation": return .dashPunctuation
+    case "pe", "closepunctuation": return .closePunctuation
+    case "pf", "finalpunctuation": return .finalPunctuation
+    case "pi", "initialpunctuation": return .initialPunctuation
+    case "po", "otherpunctuation": return .otherPunctuation
+    case "ps", "openpunctuation": return .openPunctuation
+    case "s", "symbol": return .symbol
+    case "sc", "currencysymbol": return .currencySymbol
+    case "sk", "modifiersymbol": return .modifierSymbol
+    case "sm", "mathsymbol": return .mathSymbol
+    case "so", "othersymbol": return .otherSymbol
+    case "z", "separator": return .separator
+    case "zl", "lineseparator": return .lineSeparator
+    case "zp", "paragraphseparator": return .paragraphSeparator
+    case "zs", "spaceseparator": return .spaceSeparator
```

Comment:
Hmmm, I'm not a huge fan of this generally, but I think this source formatting would be improved if the `returns` for the cases (not necessarily for `default`) lined up on the same column. WDYT?